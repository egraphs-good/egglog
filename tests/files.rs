use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
