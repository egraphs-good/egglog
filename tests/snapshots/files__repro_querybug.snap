---
source: tests/files.rs
expression: serialized
---
nodes:
  Empty-0:
    op: Empty
    children: []
    eclass: "0"
    cost: 9223372036854776000
  mylist-0:
    op: mylist
    children: []
    eclass: "1"
    cost: 9223372036854776000
  i64-5871781006564002453:
    op: "1"
    children: []
    eclass: i64-5871781006564002453
    cost: 0
  Cons-956286968014291186:
    op: Cons
    children:
      - i64-5871781006564002453
      - Empty-0
    eclass: "1"
    cost: 1
  Unit-0:
    op: ()
    children: []
    eclass: Unit-0
    cost: 0
  eq-0:
    op: eq
    children:
      - Empty-0
      - Empty-0
    eclass: Unit-0
    cost: 1
  eq-6828067974578293639:
    op: eq
    children:
      - Cons-956286968014291186
      - mylist-0
    eclass: Unit-0
    cost: 1
root_eclasses: []
class_data:
  "0":
    type: list
  "1":
    type: list
  i64-5871781006564002453:
    type: i64
  Unit-0:
    type: Unit

